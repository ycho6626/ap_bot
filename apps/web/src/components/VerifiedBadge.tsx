import { CheckCircle, AlertCircle } from 'lucide-react';
import { cn } from '@/lib/utils';

interface VerifiedBadgeProps {
  verified: boolean;
  className?: string;
}

export function VerifiedBadge({ verified, className }: VerifiedBadgeProps) {
  return (
    <div
      className={cn(
        'verified-badge',
        verified ? 'bg-success-100 text-success-800' : 'bg-warning-100 text-warning-800',
        className
      )}
    >
      {verified ? (
        <>
          <CheckCircle className='h-3 w-3 mr-1' data-testid='verified-icon' />
          Verified
        </>
      ) : (
        <>
          <AlertCircle className='h-3 w-3 mr-1' data-testid='not-verified-icon' />
          Not Verified
        </>
      )}
    </div>
  );
}
