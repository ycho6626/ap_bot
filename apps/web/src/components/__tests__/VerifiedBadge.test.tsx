import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { VerifiedBadge } from '../VerifiedBadge';

describe('VerifiedBadge', () => {
  it('should render verified badge when verified is true', () => {
    render(<VerifiedBadge verified={true} />);
    
    expect(screen.getByText('Verified')).toBeInTheDocument();
    expect(screen.getByTestId('verified-icon')).toBeInTheDocument();
  });

  it('should render not verified badge when verified is false', () => {
    render(<VerifiedBadge verified={false} />);
    
    expect(screen.getByText('Not Verified')).toBeInTheDocument();
    expect(screen.getByTestId('not-verified-icon')).toBeInTheDocument();
  });

  it('should apply custom className', () => {
    const { container } = render(<VerifiedBadge verified={true} className="custom-class" />);
    
    expect(container.firstChild).toHaveClass('custom-class');
  });

  it('should have correct ARIA attributes', () => {
    render(<VerifiedBadge verified={true} />);
    
    const badge = screen.getByText('Verified');
    expect(badge).toBeInTheDocument();
  });
});
