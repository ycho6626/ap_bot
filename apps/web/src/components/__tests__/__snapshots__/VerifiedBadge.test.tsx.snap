// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VerifiedBadge > matches snapshot for not verified state 1`] = `
<div
  class="verified-badge bg-warning-100 text-warning-800"
>
  <svg
    class="lucide lucide-alert-circle h-3 w-3 mr-1"
    data-testid="not-verified-icon"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="10"
    />
    <line
      x1="12"
      x2="12"
      y1="8"
      y2="12"
    />
    <line
      x1="12"
      x2="12.01"
      y1="16"
      y2="16"
    />
  </svg>
  Not Verified
</div>
`;

exports[`VerifiedBadge > matches snapshot for verified state 1`] = `
<div
  class="verified-badge bg-success-100 text-success-800"
>
  <svg
    class="lucide lucide-check-circle h-3 w-3 mr-1"
    data-testid="verified-icon"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
    />
    <path
      d="m9 11 3 3L22 4"
    />
  </svg>
  Verified
</div>
`;
